internal/tmpl/tmpl.go

```diff
@@ -346,3 +347,28 @@ func mdv2Escape(s string) string {
 		"!", "\\!",
 	).Replace(s)
 }
+
+type stringMap map[string]string
+
+// Get returns the value for the given key, or the default value if the key is
+// not present.
+func (m stringMap) Get(key string, def ...string) string {
```

Comment:
Can the user use the `index` template fn for this?

Reply:
They can. However, `index` returns zero value if the key is not present. If the use want to fallback to some default value, the have to use `index` combined with `or`, which may be a little cubersome. Take `name_template` for an example:
```yaml
name_template: >-
  {{ $arch := map "amd64" "x86_64" "386" "i386" -}}
  {{ .ProjectName }}_
  {{- title .OS }}_
  {{- or (index $arch .Arch) .Arch }}
  {{- if .Arm }}v{{ .Arm }}{{ end }}
```

Reply:
Do you recommend making it a method of map, or a function?
`$arch.IndexOrDefault .Arch .Arch` or `indexOrDefault $arch .Arch .Arch`?

Reply:
A function is more consistent I think